File: headlinesfeature/src/main/java/com/genrikhsalexandr/headlinesfeature/presentation/ArticleItemList.java
package com.genrikhsalexandr.headlinesfeature.presentation;

import com.genrikhsaleksandr.core.domain.model.Article;

public class ArticleItemList {
    private final String sourceName;
    private final String title;
    private final String urlToImage;
    private final Article article;
    private final long id;

        public ArticleItemList(String sourceName, String title, String urlToImage, Article article, long id) {
            this.sourceName = sourceName;
            this.title = title;
            this.urlToImage = urlToImage;
            this.article = article;
            this.id = id;
        }

        public String getSourceName() {
            return sourceName;
        }

        public String getTitle() {
            return title;
        }

        public String getUrlToImage() {
            return urlToImage;
        }

        public Article getArticle() {
            return article;
        }

        public long getId() {
            return id;
        }
}

File: core/src/main/java/com/genrikhsaleksandr/core/presentation/ArticleItemList.kt
package com.genrikhsaleksandr.core.presentation

import com.genrikhsaleksandr.core.R
import com.genrikhsaleksandr.core.domain.model.Article
import java.util.UUID

sealed interface ItemList {
    val viewType: Int
    val recyclerId: String
}

data object LoadingItemList : ItemList {

    override val viewType: Int = R.layout.list_item_loading

    override val recyclerId: String = UUID.randomUUID().toString()
}

data class ArticleItemList(
    val sourceName: String?,
    val title: String,
    val urlToImage: String?,
    val article: Article,
    val id: Long
) : ItemList {
    override val viewType: Int = R.layout.list_item_article

    override val recyclerId: String = article.url
}